lib/src/webvtt/WebVttParser.ts

```diff
@@ -0,0 +1,459 @@
+import { parseCue } from './parse/parseCue.js';
+import { parseOptions } from './parse/parseOptions.js';
+import { parseTimeStamp } from './parse/parseTimestamp.js';
+import { Settings } from './parse/Settings.js';
+import { WebVttParserState } from './parse/WebVttParserState.js';
+import type { WebVttCue } from './WebVttCue.js';
+import type { WebVttCueFactory } from './WebVttCueFactory.js';
+import type { WebVttParserOptions } from './WebVttParserOptions.js';
+import { WebVttParsingError } from './WebVttParsingError.js';
+import type { WebVttRegion } from './WebVttRegion.js';
+import type { WebVttRegionFactory } from './WebVttRegionFactory.js';
+
+const BAD_SIGNATURE = 'Malformed WebVTT signature.';
+
+/**
+ * A WebVTT parser.
+ *
+ * @group WebVTT
+ *
+ * @beta
+ *
+ * @example
+ * {@includeCode ../../test/webvtt/WebVttParser.test.ts#example}
+ */
+export class WebVttParser {
+	private state: WebVttParserState;
+	private buffer: string;
+	private regionList: WebVttRegion[];
+	private regionSettings: Settings | null = null;
+	private style: string;
+	private cue: WebVttCue | null = null;
+	private createCue: WebVttCueFactory;
+	private createRegion: WebVttRegionFactory;
+
+	/**
+	 * A callback function that is called when a parsing error occurs.
+	 */
+	onparsingerror: ((error: WebVttParsingError) => void) | undefined;
```

Comment:
Is there a reason not to use `?:` syntax, for example:

```suggestion
	onparsingerror?: (error: WebVttParsingError) => void;
```

Similar question to other `on<handler>` functions

Reply:
Fixed.